src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -397,14 +396,16 @@ protected void getChanges2(ChangeEventSourceContext context,
                         // checkpoint is the same as from_op_id, if yes then handle the tablet
                         // for split.
                         CdcSdkCheckpoint explicitCheckpoint = tabletToExplicitCheckpoint.get(part.getId());
-                        if (explicitCheckpoint != null && cp.equals(explicitCheckpoint)) {
+                        OpId lastRecordCheckpoint = offsetContext.getSourceInfo(part).lastRecordCheckpoint();
+
+                        if (explicitCheckpoint != null && (lastRecordCheckpoint == null || lastRecordCheckpoint.isLesserThanOrEqualTo(explicitCheckpoint))) {
                             // At this position, we know we have received a callback for split tablet
                             // handle tablet split and delete the tablet from the waiting list.
 
                             // Call getChanges to make sure checkpoint is set on the cdc_state table.
                             LOGGER.info("Setting explicit checkpoint is set to {}.{}", explicitCheckpoint.getTerm(), explicitCheckpoint.getIndex());
                             setCheckpointWithGetChanges(tableIdToTable.get(part.getTableId()), part,
-                                cp, explicitCheckpoint, schemaNeeded.get(part.getId()),
+                                lastRecordCheckpoint, explicitCheckpoint, schemaNeeded.get(part.getId()),
```

Comment:
This will call the `GetChanges` API by setting the `from_op_id` as `lastRecordCheckpoint`, are we sure that won't have any side effects?

Reply:
Yes, need to revert this to cp again